src/psd_replay.ts

```diff
@@ -4,6 +4,7 @@ import { createHtmlRoom, app } from "./room";
 const VERSION_TEXT = VERSION;
 
 const rooms: Record<string, string> = {};
+const savedRooms: Map<string, string> = new Map<string, string>();
```

Comment:
Ensures that rooms that are already added to list will not be duplicated and also stops functions from running if there is no need